components/NavigationDrawer/examples/BottomDrawerExample.swift

```diff
@@ -0,0 +1,353 @@
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+    super.init(coder: aDecoder)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    self.view.backgroundColor = .lightGray
+    collectionView.frame = CGRect(x: 0, y: 0, width: self.view.bounds.width,
+                             height: self.view.bounds.height)
+    collectionView.register(UICollectionViewCell.self, forCellWithReuseIdentifier: "Cell")
+    collectionView.isScrollEnabled = false
+    collectionView.delegate = self
+    collectionView.dataSource = self
+    layout.minimumLineSpacing = 0
+    layout.minimumInteritemSpacing = 0
+    self.view.addSubview(collectionView)
+  }
+
+  override func viewWillLayoutSubviews() {
+    let s = self.view.frame.size.width / 3
+    layout.itemSize = CGSize(width: s, height: s)
+  }
+
+  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    return 102
+  }
+
+  func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath)
+    let color = CGFloat(indexPath.row % 2) * 0.2 + 0.8
+    print(indexPath.item)
+    cell.backgroundColor = UIColor(white: color, alpha: 1)
+    return cell
+  }
+
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return 100
+  }
+
+  func numberOfSections(in collectionView: UICollectionView) -> Int {
+    return 1
+  }
+}
+
+class DrawerContentViewController: UIViewController {
+
+  let preferredHeight: CGFloat = 2000
+
+  override var preferredContentSize: CGSize {
+    get {
+      return CGSize(width: view.bounds.width, height: preferredHeight)
+    }
+    set {
+      super.preferredContentSize = newValue
+    }
+  }
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.backgroundColor = UIColor.purple
+  }
+}
+
+class DrawerHeaderViewController: UIViewController {
+  let preferredHeight: CGFloat = 80
+
+  override var preferredContentSize: CGSize {
+    get {
+      return CGSize(width: view.bounds.width, height: preferredHeight)
+    }
+    set {
+      super.preferredContentSize = newValue
+    }
+  }
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    super.init(coder: aDecoder)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.backgroundColor = UIColor.blue.withAlphaComponent(0.8)
+  }
+}
+
+extension DrawerHeaderViewController : MDCBottomDrawerHeader {
+
+  func updateTransitionRatio(_ transitionToTopRatio: CGFloat) {
+    // Updates the layout according to the transition to top radio.
```

Comment:
Nit: s/transition to top radio/transitionToTopRatio/

Reply:
removed, so no-op